tests/client/amo/containers/TestCategoryPage.js

```diff
@@ -0,0 +1,57 @@
+import deepcopy from 'deepcopy';
+
+import { mapStateToProps } from 'amo/containers/CategoryPage';
+
+
+describe('CategoryPage.mapStateToProps()', () => {
+  const state = {
+    api: { clientApp: 'firefox', lang: 'fr-CA' },
+    addons: {
+      ab: { slug: 'ab', name: 'ad-block' },
+      cd: { slug: 'cd', name: 'cd-block' },
+    },
+    search: {
+      filters: { addonType: 'theme', category: 'ad-block' },
+      hasSearchParams: true,
+      loading: false,
+      results: [{ slug: 'ab', name: 'ad-block' }],
+    },
+  };
+
+  it('passes the search state if the filters and state matches', () => {
+    const props = mapStateToProps(state, {
+      location: { query: {} },
+      params: {
+        addonType: 'theme',
+        clientApp: 'firefox',
+        slug: 'ad-block',
+      },
+    });
+
+    assert.deepEqual(props, {
+      hasSearchParams: true,
+      pathname: '/themes/ad-block/',
+      queryParams: { page: 1 },
+      ...state.search,
+    });
+  });
+
+  it('does not pass search state if the filters and state do not match', () => {
+    const unmatchedState = deepcopy(state);
+    unmatchedState.search.filters.addonType = 'android';
+    const props = mapStateToProps(unmatchedState, {
+      location: { query: {} },
+      params: {
+        addonType: 'theme',
+        clientApp: 'firefox',
+        slug: 'ad-block',
+      },
+    });
+
+    assert.deepEqual(props, {
+      hasSearchParams: true,
+      pathname: '/themes/ad-block/',
+      queryParams: { page: 1 },
+    });
+  });
+});
```

Comment:
I don't think that should ever happen, as it's set by `searchStart`. Am I missing a path to an empty search state?